Source/WebCore/rendering/RenderLayer.cpp

```diff
@@ -1592,6 +1593,26 @@ void RenderLayer::setAncestorChainHasVisibleDescendant()
 
 void RenderLayer::updateAncestorDependentState()
 {
+#if ENABLE(LAYER_BASED_SVG_ENGINE)
+    m_enclosingSVGResourceContainer = nullptr;
+    m_enclosingSVGHiddenContainer = nullptr;
+    auto determineSVGAncestors = [&] (const RenderElement& renderer) {
+        for (auto* ancestor = renderer.parent(); ancestor; ancestor = ancestor->parent()) {
+            if (auto* container = dynamicDowncast<RenderSVGResourceContainer>(ancestor)) {
+                m_enclosingSVGResourceContainer = container;
+                if (m_enclosingSVGHiddenContainer)
```

Comment:
I'm confused by the exit condition. We wanted to avoid the situation where both are set.
I think it is sufficient to break when either m_enclosingSVGresourceContainer/HiddenContainer is non-null.

Reply:
You are right, fixed.